app/containers/App/index.js

```diff
@@ -8,38 +8,61 @@
 
 import React from 'react';
 import Helmet from 'react-helmet';
+import { withRouter } from 'react-router';
 
 // Import the CSS reset, which HtmlWebpackPlugin transfers to the build folder
 import 'sanitize.css/sanitize.css';
 
+import ProgressBar from 'components/ProgressBar';
 import Img from 'components/Img';
 import Footer from 'components/Footer';
 import Banner from './banner-metal.jpg';
 import A from 'components/A';
 
 import styles from './styles.css';
 
-function App(props) {
-  return (
-    <div className={styles.wrapper}>
-      <Helmet
-        titleTemplate="%s - React.js Boilerplate"
-        defaultTitle="React.js Boilerplate"
-        meta={[
-          { name: 'description', content: 'A React.js Boilerplate application' },
-        ]}
-      />
-      <A className={styles.logoWrapper} href="https://twitter.com/mxstbr">
-        <Img className={styles.logo} src={Banner} alt="react-boilerplate - Logo" />
-      </A>
-      {React.Children.toArray(props.children)}
-      <Footer />
-    </div>
-  );
+export class App extends React.Component {
+
+  state = {
+    progress: -1,
+  };
+
+  componentDidMount() {
+    this.props.router.listen(() => this.setState({ progress: 0 }));
```

Comment:
if you pass a named function, you should see it pop off the call stack when transition hook is removed from listener array! (check out [createTransitionManager.js#L53](https://github.com/mjackson/history/blob/master/modules/createTransitionManager.js#L53))